### PR TITLE
Added the function of preset import and export sharing code. 

### PR DESCRIPTION
**Pull request by schlibra**
#13 #9

### CODE REVIEW

File: src/components/settings/import-and-export/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { SettingsContainer } from '@/components/ui/styled.ts'
+import { Label } from '@/components/ui/label'
+import { Button } from '@/components/ui/button.tsx'
+import { Import, Share } from 'lucide-react'
+import { TextArea } from '@radix-ui/themes'
```
Comment (zhangyu1818) [inline src/components/settings/import-and-export/index.tsx:5]: This repository uses `shadcn`, which is based on radix-ui primitives and does not use `@radix-ui/themes`. 

By the way, the `package.json` in this PR does not include `@radix-ui/themes` as a dependency, which means this PR will not run properly.

File: src/components/settings/import-and-export/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const [style, setStyle] = useStore((store) => store.styles)
+  const [showData, setShowData] = useState('')
+  return (
+    <SettingsContainer>
+      <div className={cls}>
+        <Label>{t('export data')}</Label>
+        <Button
+          size={'icon'}
+          onClick={() => {
+            setShowData(btoa(JSON.stringify(style)))
```
Comment (zhangyu1818) [inline src/components/settings/import-and-export/index.tsx:22]: Wouldn't base64 be too long? Maybe using a plain JSON string directly would be better.

**Review by zhangyu1818 — CHANGES_REQUESTED** [review]
Thank you for submitting the PR. 
Please check my comments.

**Comment by schlibra** [comment]
@zhangyu1818 Thank you for your valuable suggestions. I will revise my code and submit Pull Requests again.

**Pull request closed without merging**